lib/rubocop/ast/node/str_node.rb

```diff
@@ -8,13 +8,56 @@ module AST
     class StrNode < Node
       include BasicLiteralNode
 
+      PERCENT_LITERAL_TYPES = {
+        :% => /^%(?![qQ])/,
```

Comment:
`%%foo%`, probably to the horror of some, is a valid string. I'm not sure if any cop currently cares, just wanted to point it out. I'm not 100% on what would be a valid delimiter for these, maybe `[^a-zA-Z]` would be good enough?

Reply:
That'd still be a `%(...)` string though, wouldn't it? The extra %s replace brackets but the macro type is still a single %.

Reply:
Yes, I think so. It's in a similar vein as `%w!a!` being a valid array (hopefully no one actually writes such code).